src/Swagger/ApiDocumentationBuilder.php

```diff
@@ -0,0 +1,343 @@
+            ];
+
+            if ($description = $resourceMetadata->getDescription()) {
+                $class = [
+                    'name' => $shortName,
+                    'description' => $description,
+                    'externalDocs' => [
+                        'url' => $prefixedShortName,
+                    ],
+                ];
+            }
+
+            $attributes = $resourceMetadata->getAttributes();
+            $context = [];
+
+            if (isset($attributes['normalization_context']['groups'])) {
+                $context['serializer_groups'] = $attributes['normalization_context']['groups'];
+            }
+
+            if (isset($attributes['denormalization_context']['groups'])) {
+                $context['serializer_groups'] = isset($context['serializer_groups']) ? array_merge($context['serializer_groups'], $attributes['denormalization_context']['groups']) : $context['serializer_groups'];
+            }
+
+            foreach ($this->propertyNameCollectionFactory->create($resourceClass, $context) as $propertyName) {
+                $propertyMetadata = $this->propertyMetadataFactory->create($resourceClass, $propertyName);
+
+                if ($propertyMetadata->isIdentifier() && !$propertyMetadata->isWritable()) {
+                    continue;
+                }
+                $range = $this->getRange($propertyMetadata);
+
+                $property[$propertyName] = [
+                    'type' => $range,
+                ];
+
+                if (is_array($range)) {
+                    $property[$propertyName] = $range;
+                }
+
+                $required = [];
+
+                if ($propertyMetadata->isRequired()) {
+                    $required = array_merge($required, [$propertyName]);
+                }
+
+                if (0 !== count($required)) {
+                    $properties[$shortName]['required'] = $required;
+                }
+
+                $properties[$shortName]['type'] = 'object';
+                $properties[$shortName]['properties'] = $property;
+            }
+
+            if ($operations = $resourceMetadata->getItemOperations()) {
+                foreach ($operations as $operationName => $itemOperation) {
+                    $swaggerOperation = $this->getSwaggerOperation($resourceClass, $resourceMetadata, $operationName, $itemOperation, $prefixedShortName, false);
+                    $itemOperations['operation'] = array_merge($itemOperations['operation'], $swaggerOperation);
+                }
+            }
+
+            try {
+                $resourceClassIri = $this->iriConverter->getIriFromResourceClass($resourceClass);
+            } catch (InvalidArgumentException $e) {
+                $resourceClassIri = '/nopaths';
+            }
+            $resourceClassIri .= '/{id}';
+
+            $itemOperationsDocs[$resourceClassIri] = $itemOperations['operation'];
+            $classes[] = $class;
+        }
+
+        $doc['swagger'] = '2.0';
```

Comment:
IMO a constant is better because the version cannot be changed without patching the code.